### PR TITLE
Rename func name according TODO

### PR DESCRIPTION
**Pull request by junxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/scheduler/algorithm/priorities/metadata.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type PriorityMetadataFactory struct {
 	statefulSetLister algorithm.StatefulSetLister
```
Comment (dixudx) [inline pkg/scheduler/algorithm/priorities/metadata.go]: What is above `log` file for?
Should remove such un-used files.

**Review by dixudx — CHANGES_REQUESTED** [review]
Delete unused files.

File: pkg/scheduler/algorithm/priorities/metadata.go (hunk 1 above)
Comment (junxu) [reply pkg/scheduler/algorithm/priorities/metadata.go]: Oh，I will remove it, thx!

File: pkg/scheduler/algorithm/types.go [hunk 2]
```diff
@@ -45,8 +45,7 @@ type PredicateMetadataProducer func(pod *v1.Pod, nodeNameToInfo map[string]*sche
 
 // MetadataProducer is a function that computes metadata for a given pod. This
```
Comment (dixudx) [inline pkg/scheduler/algorithm/types.go]: Update here as well.

File: pkg/scheduler/algorithm/types_test.go [hunk 3]
```diff
@@ -24,17 +24,17 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/schedulercache"
 )
 
-// EmptyMetadataProducer should returns a no-op MetadataProducer type.
-func TestEmptyMetadataProducer(t *testing.T) {
+// EmptyPriorityMetadataProducer should returns a no-op MetadataProducer type.
```
Comment (dixudx) [inline pkg/scheduler/algorithm/types_test.go]: `returns a no-op PriorityMetadataProducer type`

File: pkg/scheduler/factory/plugins.go [hunk 4]
```diff
@@ -49,9 +49,8 @@ type PluginFactoryArgs struct {
 	HardPodAffinitySymmetricWeight int32
 }
 
-// MetadataProducerFactory produces MetadataProducer from the given args.
-// TODO: Rename this to PriorityMetadataProducerFactory.
-type MetadataProducerFactory func(PluginFactoryArgs) algorithm.MetadataProducer
+// PriorityMetadataProducerFactory produces MetadataProducer from the given args.
```
Comment (dixudx) [inline pkg/scheduler/factory/plugins.go]: sed -i s/MetadataProducer/PriorityMetadataProducer/

Comment (junxu) [reply pkg/scheduler/factory/plugins.go]: Done, thx!

**Comment by zouyee** [comment]
Please squash commit.

**Comment by k82cn** [comment]
/approve no-issue

Just a rename, I'd like to get it merged :).

**Pull request merged**